src/game/savegame/savegame_bson.c

```diff
@@ -531,6 +531,13 @@ static bool Savegame_BSON_LoadItems(struct json_array_s *items_arr)
             } else if (obj->intelligent) {
                 item->data = NULL;
             }
+
+            if (item->object_number == O_FLAME_EMITTER
+                && g_Config.enable_enhanced_saves) {
+                int16_t flame_num =
+                    json_object_get_int(item_obj, "flame_num", flame_num);
+                item->data = (void *)(flame_num + 0);
```

Comment:
Why `+ 0`?

Reply:
Oh right I meant to ask about this. I get a compiler warning without the addition that I don't remember atm. I tried to use the same formula as in [flame.c](https://github.com/rr-/Tomb1Main/blob/eeb3f4d1b2ce9e554614b6100d6175dd22bf9edd/src/game/objects/traps/flame.c#L114) but that had a `+1` in it. I'll post the warning later.

Is this solution ok btw? It's a little hacky during the load process, but the way `item->data` is used for this trap is unique.

Reply:
The compiler warning is probably about implicit type casting from int16_t to void*. Try (void*)flame_num.

Reply:
Here's the warning:
```
../src/game/savegame/savegame_bson.c: In function ‘Savegame_BSON_LoadItems’:
../src/game/savegame/savegame_bson.c:539:30: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  539 |                 item->data = (void*)flame_num;
      |                              ^
```

Reply:
Ah OK in that case just make it an `int`.